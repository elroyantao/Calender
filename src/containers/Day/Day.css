.Day {
  width: calc((99.99% - 14px) / 7);
  height: 100px;
  background: #fff;
  border: 1px solid #999;
}

.Day--empty {
  background: #999
}