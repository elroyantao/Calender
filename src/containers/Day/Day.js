import React, { Component } from 'react'
import PropTypes from 'prop-types'

if (process.browser) require('./Day.css')

class Day extends Component {
  static propTypes = {
    className: PropTypes.string,
    date: PropTypes.number
  }

  static defaultProps = {
    className: ''
  }

  render() {
    const { className, date } = this.props
    return (
      <div className={`Day ${className}`}>
        {date && <div className="Day-date">{date}</div>}
      </div>
    )
  }
}

export default Day