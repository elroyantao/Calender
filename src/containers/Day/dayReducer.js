// The initial state of the App
import { convertDateToObject } from '../../lib/date-helpers'

const today = convertDateToObject(new Date())

const initialState = {
  date: today
}

function selectedDateReducer(state = initialState, action) {
  switch (action.type) {
    case 'SET_DATE':
      return {
        ...state,
        ...action.date
      }
    default:
      return state;
  }
}

export default selectedDateReducer;