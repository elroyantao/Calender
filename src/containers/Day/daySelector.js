import { createSelector } from 'reselect'

const getReminderIds = (state, key) => state.reminder.reminderDays[key]
const getReminders = (state) => state.reminder.reminderList

export const getDaysReminders = createSelector(
  [ getReminderIds, getReminders ],
  (reminderIds = [], reminders) => reminderIds.map((reminderId) => reminders[reminderId])
    .sort((reminderA, reminderB) => reminderA.order - reminderB.order)
)