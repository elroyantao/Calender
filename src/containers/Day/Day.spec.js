import testComponentHelper from '../../lib/test-helper'
import Day from './Day'

describe('<Day />', () => {
  const renderComponent = testComponentHelper(Day)
  describe('@renders', () => {
    it('in default state',() => {
      expect(renderComponent().getTree()).toMatchSnapshot()
    })
    it('with classname', () => {
      expect(renderComponent({
        className: 'mock-class'
      }).getTree())
        .toMatchSnapshot()
    })
    it('with date', () => {
      expect(renderComponent({
        date: 14
      }).getTree())
        .toMatchSnapshot()
    })
  })
})