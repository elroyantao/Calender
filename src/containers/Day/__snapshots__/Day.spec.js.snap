// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day /> @renders in default state 1`] = `
<div
  className="Day "
/>
`;

exports[`<Day /> @renders with classname 1`] = `
<div
  className="Day mock-class"
/>
`;

exports[`<Day /> @renders with date 1`] = `
<div
  className="Day "
>
  <div
    className="Day-date"
  >
    14
  </div>
</div>
`;
